src/Raylib/Shapes/Rectangle.hpp

```diff
@@ -0,0 +1,167 @@
+}
+
+namespace raylib
+{
+    namespace shapes
+    {
+        class Rectangle {
+          public:
+            /// This function creates a rectangle with a position, width, height, and
+            /// color
+            ///
+            /// @param posX The x position of the rectangle.
+            /// @param posY The Y position of the rectangle.
+            /// @param width The width of the rectangle.
+            /// @param height The height of the rectangle.
+            /// @param color The color of the rectangle.
+            Rectangle(float posX = 0, float posY = 0, float width = 0, float height = 0, Color color = {0, 0, 0, 0});
+
+            /// A constructor that takes a Rectangle as a parameter.
+            ///
+            /// @param rec The rectangle to copy
+            Rectangle(Vector2 position, float width = 0, float height = 0, Color color = {0, 0, 0, 0});
+
+            /// This function creates a rectangle with a position, width, height, and
+            /// color
+            ///
+            /// @param pos The position of the rectangle.
+            /// @param width The width of the rectangle.
+            /// @param height The height of the rectangle.
+            /// @param color The color of the rectangle.
+            Rectangle(::Rectangle rec);
+
+            /// The destructor for the Rectangle class.
+            ~Rectangle();
+
+            /// "Return a C++ Rectangle object with the same position and dimensions as
+            /// this Rectangle object."
+            ///
+            /// The first line of the function is a return statement. It returns a C++
+            /// Rectangle object. The return statement is followed by a pair of curly
+            /// braces. The curly braces contain a list of arguments. The arguments are
+            /// separated by commas. The arguments are the same arguments that are
+            /// passed to the constructor of the C++ Rectangle class
+            ///
+            /// @return A rectangle object.
+            ::Rectangle C_rectangle();
+
+            /// `void Rectangle::draw() { DrawRectangleRec(this->C_rectangle(), _color);
+            /// }`
+            ///
+            /// The function `draw()` is a member function of the class `Rectangle`. It
+            /// is a public function, so it can be called from outside the class. It
+            /// takes no arguments and returns nothing. It calls the function
+            /// `DrawRectangleRec()` from the raylib library, which takes two arguments:
+            /// a `Rectangle` and a `Color`. The first argument is the `Rectangle` that
+            /// is a member of the class `Rectangle`. The second argument is the `Color`
+            /// that is a member of the class `Rectangle`
+            void draw();
+
+            /// Draws a rectangle outline (or "wireframe") with the given color
+            ///
+            /// @param color The color of the rectangle.
+            void drawLines(Color color);
```

Comment:
Same comment as for circle, the class Rectangle already has a _color attribute, use it instead of asking it for each draw.